server/game/GameActions/DestroyAction.js

```diff
@@ -24,30 +24,50 @@ class DestroyAction extends CardGameAction {
-            event.destroyEvent = context.game.getEvent('onCardDestroyed', {
+
+            event.leavesPlayEvent = context.game.getEvent('onCardLeavesPlay', {
                 card: event.card,
                 context: context,
                 condition: event => event.card.location === 'play area',
-                isFullyResolved: event => !!event.leavesPlayEvent && event.leavesPlayEvent.isFullyResolved(event.leavesPlayEvent),
-                inFight: event.inFight,
+                triggeringEvent: event,
                 battlelineIndex: event.card.controller.creaturesInPlay.indexOf(event.card) - 1
-            }, event => {
-                event.leavesPlayEvent = context.game.getEvent('onCardLeavesPlay', {
-                    card: event.card,
-                    context: event.context,
-                    battlelineIndex: event.card.controller.creaturesInPlay.indexOf(event.card) - 1
-                }, event => {
-                    event.card.owner.moveCard(event.card, this.purge ? 'purged' : 'discard');
-                });
-                event.addSubEvent(event.leavesPlayEvent);
+            }, leavesPlayEvent => {
+                if(context.game.firstDestroyEvent === leavesPlayEvent) {
+                    context.game.firstDestroyEvent = null;
+                } else if(!this.purge && context.game.firstDestroyEvent && !context.game.firstDestroyEvent.getChildEvents().includes(leavesPlayEvent)) {
+                    let newDestroyEvent = super.createEvent('unnamedEvent', {
+                        card: event.card,
+                        context: event.context,
+                        damageEvent: event.damageEvent
+                    }, newEvent => newEvent.name = 'onCardDestroyed');
+                    context.game.firstDestroyEvent.addChildEvent(newDestroyEvent);
+                    context.game.firstDestroyEvent.addChildEvent(context.game.getEvent('unnamedEvent', {
+                        card: leavesPlayEvent.card,
+                        context: leavesPlayEvent.context,
+                        condition: event => event.card.location === 'play area',
+                        triggeringEvent: newDestroyEvent,
+                        battlelineIndex: leavesPlayEvent.battlelineIndex
+                    }, newEvent => {
+                        newEvent.name = 'onCardLeavesPlay';
+                        newEvent.card.owner.moveCard(event.card, this.purge ? 'purged' : 'discard');
+                    }));
+                    return;
+                }
+
+                leavesPlayEvent.card.owner.moveCard(event.card, this.purge ? 'purged' : 'discard');
             });
+
             card.owner.creatureDestroyed = true;
-            event.addSubEvent(event.destroyEvent);
+            event.addSubEvent(event.leavesPlayEvent);
+            /*
+            if(!context.game.firstDestroyEvent || context.game.firstDestroyEvent.getSimultaneousEvents().every(e => e.cancelled || e.resolved)) {
```

Comment:
Is this needed?

Reply:
Yes - it needs uncommenting at some future point, but I'd rather merge what we've got before adding additional complications...